lib/msal-common/src/logger/Logger.ts

```diff
@@ -62,10 +62,16 @@ export class Logger {
         const defaultLoggerCallback = () => {
             return;
         };
-        this.localCallback = loggerOptions.loggerCallback || defaultLoggerCallback;
-        this.piiLoggingEnabled = loggerOptions.piiLoggingEnabled || false;
-        this.level = typeof(loggerOptions.logLevel) === "number" ? loggerOptions.logLevel : LogLevel.Info;
-        this.correlationId = loggerOptions.correlationId || Constants.EMPTY_STRING;
+        const defaultLoggerOptions = {};
+        if(!loggerOptions)
+        {
+        // eslint-disable-next-line no-param-reassign
+            loggerOptions= defaultLoggerOptions;
+        }
+        this.localCallback = loggerOptions?.loggerCallback || defaultLoggerCallback;
```

Comment:
Logger options should be defined by now, '?' should not be needed

Reply:
When the user is setting the logger options in this way:-
{
loggerOptions:
           void 0
}
the sample throws the undefined error, its not using defaultCallback. Hence , had to make these as optional.

Reply:
void 0 is not an accepted type therefore it's expected that this would fail

Reply:
By the time the Logger is initialized we should have a defined logger options object that can be passed in (whether it is user provided or defaults) but that logic lives further up the stack (in msal-browser/node) which is why it is a required parameter here